signatures.md

```diff
@@ -52,163 +49,359 @@ Note: `Message` is used here as the object being signed, but this interface shou
+        //
+        Verify(smsg SignedMessgage, pk PublicKey) (valid bool, err error)
+  
+  			// VerifyAggregate validates a collection of statements that are represeted by 
+  			// a signature aggregate: "For each M_i and PK_i in smsgs, only this collection 
+  			// of {M}, {PK}, and signers could have generated `smsgs.SignatureBytes`
+  			//
+        // Out:
+        //   valid - a boolean value indicating the signature is valid
+        //   err - a standard error message indicating any process issues
         // In:
-        //    m - a series of bytes representing the signed message
-        //    pk - the public key belonging to the signer `M`
-        //    sig - a series of bytes representing a signature usually `r`|`s`
+        //   smsgs - a SignedMessage which contains {[]Message, []Address, SignatureBytes}
+        //   pks - an array of Public Keys of the appropriate type associated with a message
+  			// 	 at index `i` in []Message.
         //
-        Verify(m Messgage, pk PublicKey, sig SignatureBytes) (valid bool, err error)
+  			VerifyAggregate(smsgs SignedMessgage,  pks []PublicKeys) (valid bool, err error)
 
-        // Recover, as its name implies, recovers a public key associated with a 
-        // particular signature. In the case of ECDSA signatures, this function can 
-        // be fulfilled via the 'ECRecover' method. If a different signature scheme 
-        // is used, then some other mechanism of 'recovering' a message authors 
-        // public key must be provided.
+        // Recover determines the public key associated with a particular signature. For
+  			// ECDSA signatures public keys can be recovered from SignatureBytes and an
+  			// associated message. For BLS signatures, public keys will be recovered by lookup 
+  			// using a mapping from Address to public key.
         //
         // Out:
-        //    pk - the public key associated with `M` who signed `m`
+        //    pk - the public key associated with `M` who signed `msg`
         //    err - a standard error message indicating any process issues
         //    **
         // In:
-        //    m - a series of bytes representing the signed message
-        //    sig - a series of bytes representing a signature usually `r`|`s`
+        //    msg - a series of bytes representing the signed message
+        //    sig - a series of bytes representing a signature
         //
-        Recover(m Message, sig SignatureBytes) (pk PublicKey, err error)
+        Recover(msg Message, addr Address, sig SignatureBytes) (pk PublicKey, err error)
 }
 ```
 
+### Signed Message
 
-# Selected Signature Scheme
+This is the container for a signature that combines a message with an address and signature. 
 
-Currently, Filecoin uses secp256k1 signatures to fulfill the above interface. All signatures on messages, blocks, and tickets currently use the same scheme and format.
+```go
+Type SignedMessage Struct {
```

Comment:
where does this type actually get used?  This is not the signed message that gets sent over the network for message propogation, its something else, right?

Reply:
There should be only one `SignedMessage` --> https://github.com/filecoin-project/go-filecoin/blob/master/types/signed_message.go#L22-L28

If there are more than one, then we should do some disambiguation.